extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/common/KubernetesPeonClient.java

```diff
@@ -274,6 +279,14 @@ private void emitK8sPodMetrics(Task task, String metric, long durationMs)
   {
     ServiceMetricEvent.Builder metricBuilder = new ServiceMetricEvent.Builder();
     IndexTaskUtils.setTaskDimensions(metricBuilder, task);
+    ExecutionConfig executionConfig = executionConfigRef.get();
+    if (executionConfig != null && executionConfig.getBehaviorStrategy() != null) {
```

Comment:
would this ever actually be null? with the default can't we set a @Nonnull flag on it and assume it'll be there?